### PR TITLE
354236 - Bounding box with markers per search result with geospatial data

### PR DESCRIPTION
**Pull request by FujaleAhmedShaik**
### What one thing this PR does?

The search result bounding boxes can now be seen on a map. Each bounding box would have a marker positioned at the center of  box.

### Task details

- [354236 - Display markers for each search result with geospatial data.](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/354236)

### Other notes

- We have a checkbox under filter sections. This checkbox allows us to toggle the bounding box.
- By default, the checkbox will be checked and the bounding boxes will appear on the screen.
- When we exit the map, the checkbox state will return to checked if it was unchecked.

### How do these changes look like?

**Initial Map render with boxes**
| Initial Map render with boxes |
| ------------------------------ |

**Toggle the boxes**
| Toggle the boxes |
| ------------------------------ |

### Dev-tested in

1. Local environment

- [Search results](http://localhost:3000/search)

2. Dev environment

- [Search results](https://dev-ncea-search.azure.defra.cloud/search)

### CODE REVIEW

File: src/utils/formatSearchResponse.ts [hunk 1]
```diff
@@ -48,6 +55,17 @@ const formatSearchResponse = async (
...
     };
 
+    if (isMapResults) {
+      const coordinatesData: IAccumulatedCoordinatesWithCenter | null = getAccumulatedCoordinatesNCenter(
+        searchItem?._source?.geom ?? {},
+      );
+      item = {
+        ...item,
+        geographicBoundary: coordinatesData?.coordinates,
+        geographicCenter: coordinatesData?.center,
```
Comment (karthigayayiniram) [inline src/utils/formatSearchResponse.ts:65]: geographicCenter & geographicBoundary value may be undefined. but at location.js line 277 you are using it. location.js works with undefined values?

File: src/utils/transformErrors.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ const dateErrorHandler = (error: Joi.ValidationError): FormFieldError | undefine
   let toError: string = '';
   const toItems: GovUKItems[] = [];
   Object.keys(error._original).forEach((field) => {
+    console.log(field);
```
Comment (karthigayayiniram) [inline src/utils/transformErrors.ts:63]: do you need this? if not plz remove it

File: src/controllers/web/SearchResultsController.ts [hunk 3]
```diff
@@ -90,6 +97,7 @@ const SearchResultsController = {
         detailsTabOptions,
       });
     } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (FujaleAhmedShaik) [reply src/controllers/web/SearchResultsController.ts:100]: This has been removed now.

File: src/utils/transformErrors.ts (hunk 2 above)
Comment (FujaleAhmedShaik) [reply src/utils/transformErrors.ts:63]: This has been removed now.

File: src/utils/formatSearchResponse.ts (hunk 1 above)
Comment (FujaleAhmedShaik) [reply src/utils/formatSearchResponse.ts:65]: For map results, we use an ES query to identify the record that contains geocoordinate data. There is no possibility of undefined values for coordinates and center.

**Review by karthigayayiniram — APPROVED** [review]

**Review by ganesh-rajamahanthy — APPROVED** [review]

**Pull request merged**